Dockerfile

```
@@ -1,7 +1,40 @@
-FROM ubuntu:latest
-WORKDIR /app
-COPY ./target/release/prover-server .
-COPY ./libzktrie.so .
-ENV LD_LIBRARY_PATH=/app/libzktrie.so:
+FROM zktachyon/halo2:jammy-latest AS builder
+LABEL maintainer="Kroma <kroma@kroma.network>"
+
+ARG RUST_TOOLCHAIN_VERSION=nightly-2022-12-10
+
+RUN apt update && \
+    apt install -y --no-install-recommends \
+    build-essential \
+    ca-certificates \
+    curl \
+    libclang-dev && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --default-toolchain ${RUST_TOOLCHAIN_VERSION} -y
+ENV PATH="/root/.cargo/bin:${PATH}"
```

It's very minor but it would be great if you match this with the below `ENV PATH=`

```suggestion
ENV PATH="${PATH}:/root/.cargo/bin"
```